explain-aggregation.md

```diff
@@ -0,0 +1,172 @@
+|   └─StreamAgg_8            | 1.00      | 4       | cop[tikv] |               | proc max:12ms, min:12ms, p80:12ms, p95:12ms, iters:122, tasks:4                                                                                                                                                                                 | funcs:count(1)->Column#7        | N/A       | N/A  |
+|     └─TableFullScan_15     | 242020.00 | 121010  | cop[tikv] | table:t1      | proc max:12ms, min:12ms, p80:12ms, p95:12ms, iters:122, tasks:4                                                                                                                                                                                 | keep order:false                | N/A       | N/A  |
++----------------------------+-----------+---------+-----------+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------+-----------+------+
+4 rows in set (0.01 sec)
+```
+
+## Hash Aggregation
+
+The Hash Aggregation algorithm uses a hash table to store intermediate results while performing aggregation. It executes in parallel using multiple threads, but consumes more memory than stream aggregation.
+
+The following is an example of the `HashAgg` operator:
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT /*+ HASH_AGG() */ count(*) FROM t1;
+```
+
+```sql
++---------------------------+-----------+-----------+---------------+---------------------------------+
+| id                        | estRows   | task      | access object | operator info                   |
++---------------------------+-----------+-----------+---------------+---------------------------------+
+| HashAgg_9                 | 1.00      | root      |               | funcs:count(Column#6)->Column#5 |
+| └─TableReader_10          | 1.00      | root      |               | data:HashAgg_5                  |
+|   └─HashAgg_5             | 1.00      | cop[tikv] |               | funcs:count(1)->Column#6        |
+|     └─TableFullScan_8     | 242020.00 | cop[tikv] | table:t1      | keep order:false                |
++---------------------------+-----------+-----------+---------------+---------------------------------+
+4 rows in set (0.00 sec)
+```
+
+The `operator info` shows that the hashing function used to aggregate the data is `funcs:count(1)->Column#6`.
+
+## Stream Aggregation
+
+The Stream Aggregation algorithm usually takes less memory than Hash Aggregation. However, this operator requires that data is sent ordered so that it can _stream_ and apply the aggregation on values as they arrive.
```

Comment:
```suggestion
The Stream Aggregation algorithm usually consumes less memory than Hash Aggregation. However, this operator requires that data is sent ordered so that it can _stream_ and apply the aggregation on values as they arrive.
```